unit/java_bytecode/java_bytecode_parse_generics/parse_generic_class_with_generic_inner_classes.cpp

```diff
@@ -0,0 +1,918 @@
+        require_type::require_component(java_generic_class, "field5");
+
+      require_type::require_pointer(
+        field_component.type(),
+        symbol_typet(
+          class_prefix + "$GenericInnerClass$DoublyNestedInnerClass"));
+
+      THEN("The pointer should be generic")
+      {
+        require_type::require_java_generic_type(
+          field_component.type(),
+          {{require_type::type_parameter_kindt::Var, class_prefix + "::T"},
+           {require_type::type_parameter_kindt::Var, class_prefix + "::T"}});
+      }
+    }
+
+    THEN(
+      "The field6 component should be a pointer to GenericClassWithGeneric"
+      "InnerClasses$GenericInnerClass$DoublyNestedInnerGenericClass")
+    {
+      const struct_typet::componentt &field_component =
+        require_type::require_component(java_generic_class, "field6");
+
+      require_type::require_pointer(
+        field_component.type(),
+        symbol_typet(
+          class_prefix + "$GenericInnerClass$DoublyNestedInnerGenericClass"));
+
+      THEN("The pointer should be generic")
+      {
+        require_type::require_java_generic_type(
+          field_component.type(),
+          {{require_type::type_parameter_kindt::Var, class_prefix + "::T"},
+           {require_type::type_parameter_kindt::Inst,
+            "java::Interface_Implementation"},
+           {require_type::type_parameter_kindt::Inst,
+            "java::Interface_Implementation"}});
+      }
+    }
+
+    THEN(
+      "The field7 component should be a pointer to GenericClassWithGeneric"
+      "InnerClasses$GenericInnerClass$DoublyNestedInnerGenericClass")
+    {
+      const struct_typet::componentt &field_component =
+        require_type::require_component(java_generic_class, "field7");
+
+      require_type::require_pointer(
+        field_component.type(),
+        symbol_typet(
+          class_prefix + "$GenericInnerClass$DoublyNestedInnerGenericClass"));
+
+      THEN("The pointer should be generic")
+      {
+        require_type::require_java_generic_type(
+          field_component.type(),
+          {{require_type::type_parameter_kindt::Var, class_prefix + "::T"},
+           {require_type::type_parameter_kindt::Var, class_prefix + "::T"},
+           {require_type::type_parameter_kindt::Var, class_prefix + "::T"}});
+      }
+    }
+
+    THEN(
+      "The field8 component should be a pointer to "
+      "GenericClassWithGenericInnerClasses$TwoParamInnerClass")
+    {
+      const struct_typet::componentt &field_component =
+        require_type::require_component(java_generic_class, "field8");
+
+      require_type::require_pointer(
+        field_component.type(),
+        symbol_typet(class_prefix + "$TwoParamInnerClass"));
+
+      THEN("The pointer should be generic")
+      {
+        require_type::require_java_generic_type(
+          field_component.type(),
+          {{require_type::type_parameter_kindt::Var, class_prefix + "::T"},
```

Comment:
Same minor nit here about making the layout of all three elements of the initializer list be formatted in the same layout.